src/nvim/channel.h

```diff
@@ -137,5 +153,41 @@ static inline Stream *channel_outstream(Channel *chan)
   abort();
 }
 
+static inline Channel *acquire_asynccall_channel(void)
+{
+  typval_T jobid_tv = TV_INITIAL_VALUE;
+  executor_exec_lua(
+      STATIC_CSTR_AS_STRING("return vim._create_nvim_job()"),
+      &jobid_tv);
+  return find_channel((uint64_t)jobid_tv.vval.v_number);
+}
+
+static inline void release_asynccall_channel(Channel *channel)
+{
+  process_stop((Process *)&channel->stream.proc);
+}
+
+static inline void put_result(uint64_t job, Object result, Error *err)
+{
+  Array args = ARRAY_DICT_INIT;
+  ADD(args, INTEGER_OBJ((long)job));
+  ADD(args, copy_object(result));
+  nvim_execute_lua(
+      STATIC_CSTR_AS_STRING("return vim._put_result(select(1, ...))"),
```

Comment:
Whenever the return value of (c function) `nvim_execute_lua` is not used, we should drop `return ` from the lua statement. This ensures an allocated API value cannot be leaked, even if the lua code did something unexpected.

Reply:
also, if `nvim_execute_lua(STATIC_CSTR_AS_STRING("..."), ...)` becomes common (which is a good thing), maybe it can be a dedicated macro: `EXEC_LUA_STATIC("vim._put_result(...)", args, err)`